internal/manifest/v1beta1/spec_resolver.go

```diff
@@ -174,27 +172,28 @@ func (m *ManifestSpecResolver) getValuesFromConfig(
 }
 
 var (
-	ErrChartConfigObjectInvalid = errors.New("chart config object of .spec.config is invalid")
-	ErrConfigObjectInvalid      = errors.New(".spec.config is invalid")
+	ErrConfigObjectInvalid = errors.New(".spec.config is invalid")
 )
 
-func parseInstallConfigs(decodedConfig interface{}) ([]interface{}, error) {
+func ParseInstallConfigs(decodedConfig interface{}) ([]interface{}, error) {
 	var configs []interface{}
-	installConfigObj, decodeOk := decodedConfig.(map[string]interface{})
-	if !decodeOk {
+
+	type ConfigsYaml struct {
```

Comment:
Define a struct inside function in general not a best practice, for example, you immediately lost the benefits to implement function for this struct, although in this case, you don't need it.